portal/src/components/Header/HeaderContext.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { createContext, useReducer, Dispatch } from "react";
+import { hamburgerReducer, HamburgerActions } from "./HeaderReducers";
+
+type InitialStateType = {
+    isOpenBurger: boolean;
+};
+
+const initialState = {
+    isOpenBurger: false,
+};
+
+const HeaderContext = createContext<{
+    state: InitialStateType;
+    dispatch: Dispatch<HamburgerActions>;
+}>({
+    state: initialState,
+    dispatch: () => null,
+});
+
+const mainReducer = ({ isOpenBurger }: InitialStateType, action: HamburgerActions) => ({
+    isOpenBurger: hamburgerReducer(isOpenBurger, action),
+});
+
+const HeaderProvider: React.FC = ({ children }) => {
```

Comment:
Hvis denne kun håndterer state for hamburgeren ville jeg kanskje foretrukket `HamburgerProvider` e.l. som navn, og heller endre navn om vi får inn mer state som angår headeren.

Reply:
Burde ikke `HamburgerProvider` være `McDonals` eller `BurgerKing`?